File: my-submissions/0128-longest-consecutive-sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numset(nums.begin(),nums.end());

        int maxlen = 0;

        for(int num :numset){

            if(numset.find(num-1) == numset.end()){
                int len = 1;
                int currnum = num;
                while(numset.count(currnum+1)){
                    len++;
                    currnum++;
                }
                maxlen = max(maxlen,len);
            }
        }
        return maxlen;
    }
};

File: my-submissions/0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int longg = 0;
        HashSet<Integer> set = new HashSet<>();

        for(int i : nums){
            set.add(i);
        }

        for(int i : set){
            if(!set.contains(i - 1)){
                int curr = i;
                int curr_st = 0 ;
                while(set.contains(curr)){
                    curr_st++;
                    curr++;
                }
                longg = Math.max(longg,curr_st);
            }
        }
        return longg;
    }
}
